spec/lang/intrinsic.md

```diff
@@ -300,3 +300,56 @@ impl<M: Memory> Machine<M> {
     }
 }
 ```
+
+Some more complex atomic memory operations are needed as well.
+
+```rust
+impl<M: Memory> Machine<M> {
```

Comment:
```suggestion

```
no reason to have these in separate blocks, they all together form the atomic memory operations